src/hotspot/share/c1/c1_InstructionPrinter.cpp

```diff
@@ -624,62 +624,44 @@ void InstructionPrinter::do_BlockBegin(BlockBegin* x) {
     return;
   }
 
-  // print phi functions
-  bool has_phis_in_locals = false;
-  bool has_phis_on_stack = false;
-
-  if (x->end() && x->end()->state()) {
-    ValueStack* state = x->state();
-
-    int i = 0;
-    while (!has_phis_on_stack && i < state->stack_size()) {
-      Value v = state->stack_at_inc(i);
-      has_phis_on_stack = is_phi_of_block(v, x);
-    }
-
-    do {
-      for (i = 0; !has_phis_in_locals && i < state->locals_size();) {
-        Value v = state->local_at(i);
-        has_phis_in_locals = is_phi_of_block(v, x);
-        // also ignore illegal HiWords
-        if (v && !v->type()->is_illegal()) i += v->type()->size(); else i ++;
-      }
-      state = state->caller_state();
-    } while (state != NULL);
-
+  if (!x->end() || !x->end()->state()) {
+    return;
   }
 
-  // print values in locals
-  if (has_phis_in_locals) {
-    output()->cr(); output()->print_cr("Locals:");
-
-    ValueStack* state = x->state();
-    do {
-      for (int i = 0; i < state->locals_size();) {
-        Value v = state->local_at(i);
-        if (v) {
-          print_phi(i, v, x); output()->cr();
-          // also ignore illegal HiWords
-          i += (v->type()->is_illegal() ? 1 : v->type()->size());
-        } else {
-          i ++;
+  // print values in locals related to phi functions
+  ValueStack* state = x->state();
+  bool printed_phis_in_locals_header = false;
+  do {
+    for (int i = 0; i < state->locals_size();) {
+      Value v = state->local_at(i);
+      if (v && is_phi_of_block(v, x)) {
+        if (!printed_phis_in_locals_header) {
+          output()->cr(); output()->print("Values in locals related to phi functions:");
```

Comment:
Can we keep this old style? i.e. "Locals:" and "Stacks:".

Reply:
Thanks @kelthuzadx 
As this patch is modifying some output content, I'm not quite sure if the labels should be modified accordingly. 
I do understand your concerns, Let's see how others think about it.